packages/nouns-contracts/contracts/NounsArt.sol

```diff
@@ -0,0 +1,449 @@
+     * format is expected to be like {encodedCompressed}: bytes created by taking a string array of
+     * RLE-encoded images, abi encoding it as a bytes array, and finally compressing it using deflate.
+     * @param decompressedLength the size in bytes the images bytes were prior to compression; required input for Inflate.
+     * @param imageCount the number of images in this batch; used when searching for images among batches.
+     * @dev This function can only be called by the descriptor.
+     */
+    function addGlassesFromPointer(
+        address pointer,
+        uint80 decompressedLength,
+        uint16 imageCount
+    ) external override onlyDescriptor {
+        addPage(glassesTrait, pointer, decompressedLength, imageCount);
+
+        emit GlassesAdded(imageCount);
+    }
+
+    /**
+     * @notice Get the number of available Noun `backgrounds`.
+     */
+    function backgroundsCount() public view override returns (uint256) {
+        return backgrounds.length;
+    }
+
+    /**
+     * @notice Get a head image bytes (RLE-encoded).
+     */
+    function heads(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(headsTrait, index);
+    }
+
+    /**
+     * @notice Get a body image bytes (RLE-encoded).
+     */
+    function bodies(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(bodiesTrait, index);
+    }
+
+    /**
+     * @notice Get a accessory image bytes (RLE-encoded).
+     */
+    function accessories(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(accessoriesTrait, index);
+    }
+
+    /**
+     * @notice Get a glasses image bytes (RLE-encoded).
+     */
+    function glasses(uint256 index) public view override returns (bytes memory) {
+        return imageByIndex(glassesTrait, index);
+    }
+
+    /**
+     * @notice Get a color palette bytes.
+     */
+    function palettes(uint8 paletteIndex) public view override returns (bytes memory) {
+        address pointer = palettesPointers[paletteIndex];
+        if (pointer == address(0)) {
+            revert PaletteNotFound();
+        }
+        return SSTORE2.read(palettesPointers[paletteIndex]);
+    }
+
+    /**
+     * @notice Decompresses Deflated bytes using the Puff algorithm
+     * based on Based on https://github.com/adlerjohn/inflate-sol.
+     * @param source the bytes to decompress.
+     * @param destlen the length of the original decompressed bytes.
+     * @return Inflate.ErrorCode 0 if successful, otherwise an error code specifying the reason for failure.
+     * @return bytes the decompressed bytes.
+     */
+    function puff(bytes memory source, uint256 destlen) public pure returns (Inflate.ErrorCode, bytes memory) {
+        return Inflate.puff(source, destlen);
+    }
+
+    function _addBackground(string calldata _background) internal {
```

Comment:
Any reason this internal function is prefixed with an underscore, while the others are not?